iterative_assembly.wdl

```diff
@@ -0,0 +1,110 @@
+# This workflow can be run on its own spades.contigs or spades.scaffolds output
+workflow iterativeAssembly {
+    File inputAssembly
+    File read1
+    File? read2
+    Int? subsampleNumber
+    Int? subsampleSeed
+    String outputDir
+
+    # First index the assembly
+    call bwa.index as bwaIndex {
+        input:
+            fasta = inputAssembly,
+            outputDir = outputDir + "/bwaIndex"
+    }
+
+    # Map the reads back to the assembly.
+    call bwa.mem as bwaMem {
+        input:
+            inputR1 = read1,
+            inputR2 = read2,
+            referenceFasta = bwaIndex.indexedFasta,
+            indexFiles = bwaIndex.indexFiles,
+            outputPath= outputDir + "/" + "ReadsMappedToInputAssembly.bam"
+    }
+
+    # Get the reads that mapped to the assembly. This means filtering out the UNMAP flag.
+    # QCFAIL is also filtered out. For paired reads, the PROPER_PAIR flag is used.
+    call samtools.view as selectMappedReads {
+        input:
+            inFile= bwaMem.bamFile,
+            outputBam = true,
+            excludeSpecificFilter = 12, # UNMAP,MUNMAP should both be present for the read to be filtered out.
+            outputFileName = outputDir + "/unmapppedReadsFiltered.bam"
+    }
+
+    call picard.SamToFastq as SamToFastq {
+        input:
+            inputBam = selectMappedReads.outputFile,
+            outputRead1 = outputDir + "/filtered_reads1.fq.gz",
+            outputRead2 = if defined(read2) then outputDir + "/filtered_reads2.fq.gz" else read2
+    }
+
+    # Allow subsampling in case number of mapped reads is too much for the assembler to handle.
+    if (defined(subsampleNumber)) {
+        call seqtk.sample as subsampleReads1 {
+            input:
+                sequenceFile = SamToFastq.read1,
+                number = subsampleNumber,
+                seed = subsampleSeed,
+                outFilePath = outputDir + "/subsampling/subsampledReads1.fq.gz",
+                zip=true
+        }
+        if (defined(read2)) {
+                    call seqtk.sample as subsampleReads2 {
+                        input:
+                            sequenceFile = select_first([SamToFastq.read2]),
+                            number = subsampleNumber,
+                            seed = subsampleSeed,
+                            outFilePath = outputDir + "/subsampling/subsampledReads2.fq.gz",
+                            zip=true
+                    }
+        }
+    }
+
+    # Make sure subsampledReads are used if subsampling was used. Default to selectedReads
+    File subsampledReads1 = select_first([subsampleReads1.subsampledReads, SamToFastq.read1])
+    File? subsampledReads2 = if defined(subsampleNumber) then subsampleReads2.subsampledReads else SamToFastq.read2
+
+    call spades.spades {
+        input:
+            read1 = subsampledReads1,
+            read2 = subsampledReads2,
+            outputDir = outputDir + "/spades"
```

Comment:
Does spades has an option to use the first assembly as a guide? This maybe can help the assembly.

Reply:
No, unfortunately